apps/emqx_authn/src/simple_authn/emqx_authn_mongodb.erl

```diff
@@ -141,29 +141,23 @@ authenticate(#{password := Password} = Credential,
               , selector := Selector0
               , '_unique' := Unique
               } = State) ->
-    try
-        Selector1 = replace_placeholders(Selector0, Credential),
-        Selector2 = normalize_selector(Selector1),
-        case emqx_resource:query(Unique, {find_one, Collection, Selector2, #{}}) of
-            undefined -> ignore;
-            {error, Reason} ->
-                ?LOG(error, "['~s'] Query failed: ~p", [Unique, Reason]),
-                ignore;
-            Doc ->
-                case check_password(Password, Doc, State) of
-                    ok ->
-                        {ok, #{is_superuser => is_superuser(Doc, State)}};
-                    {error, {cannot_find_password_hash_field, PasswordHashField}} ->
-                        ?LOG(error, "['~s'] Can't find password hash field: ~s", [Unique, PasswordHashField]),
-                        {error, bad_username_or_password};
-                    {error, Reason} ->
-                        {error, Reason}
-                end
-        end
-    catch
-        error:Error ->
-            ?LOG(warning, "The following error occurred in '~s' during authentication: ~p", [Unique, Error]),
-            ignore
+    Selector1 = replace_placeholders(Selector0, Credential),
+    Selector2 = normalize_selector(Selector1),
+    case emqx_resource:query(Unique, {find_one, Collection, Selector2, #{}}) of
+        undefined -> ignore;
+        {error, Reason} ->
+            ?LOG(error, "['~s'] Query failed: ~p", [Unique, Reason]),
+            ignore;
+        Doc ->
+            case check_password(Password, Doc, State) of
+                ok ->
+                    {ok, #{is_superuser => is_superuser(Doc, State)}};
+                {error, {cannot_find_password_hash_field, PasswordHashField}} ->
```

Comment:
why should there be a special treat for mongodb ?

Reply:
add a comment to clarify.

Reply:
I don’t think this is a special treatment. Originally, each authentication should have its own processing code, but now the other processing codes look very similar.

Reply:
mongodb is different because the configuration items it provides are also different from other authentication, so there are different errors or the processing is normal.

Reply:
then maybe add below comment here
```
%% mongodb's bad_username_or_password is different from other auth providers
```